packages/demo-protobuf/src/registry.ts

```diff
@@ -0,0 +1,27 @@
+import protobuf from "protobufjs";
+
+import { cosmos_sdk as cosmosSdk, google } from "./generated/codecimpl";
+
+abstract class GeneratedType {
+  public abstract create(properties?: { [k: string]: any }): any;
+  public abstract encode(message: any | { [k: string]: any }, writer?: protobuf.Writer): protobuf.Writer;
+  public abstract decode(reader: protobuf.Reader | Uint8Array, length?: number): any;
+}
```

Comment:
```ts
interface GeneratedType {
  readonly create: (properties?: { [k: string]: any }) => any;
  readonly encode: (message: any | { [k: string]: any }, writer?: protobuf.Writer) => protobuf.Writer;
  readonly decode: (reader: protobuf.Reader | Uint8Array, length?: number) => any;
}
```

works as well. For the generated types like `cosmosSdk.v1.Coin` this is the static interface and for the dynamic types it applies to instances of `protobuf.Type`.

Reply:
Huh, I tried something like this and it didn't like it. Not sure what was different.